README.md

```diff
@@ -66,6 +66,50 @@ Pickling a function interactively defined in a Python shell session
 85
 ```
 
+
+Overriding pickle's serialization mechanism for importable constructs:
+----------------------------------------------------------------------
+
+An important difference between `cloudpickle` and `pickle` is that
+`cloudpickle` can serialize a function or class **by value**, whereas `pickle`
+can only serialize it **by reference**, e.g. by serializing its *module
+attribute path* (such as `my_module.my_function`).
+
+By default, `cloudpickle` only uses serialization by value in cases where
+serialization by reference is usually ineffective, for example when the
+function/class to be pickled was constructed in an interactive Python session.
+
+Since `cloudpickle 1.7.0`, it is possible to extend the use of serialization by
+value to functions or classes coming from **any pure Python module**. This feature
+is useful when the said module is unavailable in the unpickling environment
+(making traditional serialization by reference ineffective). To this end,
```

Comment:
Maybe the motivation (interactive dev on a distributed cluster with long running worker processes) could be made more explicit by reusing the text I suggest in the docstring of `register_pickle_by_value` above in this review.